examples/conversation_with_RAG_agents/configs/agent_config.json

```diff
@@ -0,0 +1,72 @@
+[
+  {
+    "class": "LlamaIndexAgent",
+    "args": {
+      "name": "Tutorial-Assistant",
+      "description": "Tutorial-Assistant is an agent that can provide answer based on English tutorial material, mainly the markdown files. It can answer general questions about AgentScope.",
+      "sys_prompt": "You're an assistant helping new users to use AgentScope. The language style is helpful and cheerful. You generate answers based on the provided context. The answer is expected to be no longer than 100 words. If the key words of the question can be found in the provided context, the answer should contain the section name which contains the answer. For example, 'You may refer to SECTION_NAME for more details.'",
+      "model_config_name": "qwen_config",
+      "rag_config": {
+            "knowledge_id": ["agentscope_tutorial_rag"],
+            "similarity_top_k": 5,
+            "log_retrieval": false,
+            "recent_n_mem": 1
+      }
+    }
+  },
+  {
+    "class": "LlamaIndexAgent",
+    "args": {
+      "name": "Code-Search-Assistant",
+      "description": "Code-Search-Assistant is an agent that can provide answer based on AgentScope code base. It can answer questions about specific modules in AgentScope.",
+      "sys_prompt": "You're a coding assistant of AgentScope. The answer starts with appreciation for the question, then provide details regarding the functionality and features of the modules mentioned in the question. The language should be in a professional and simple style. The answer is limited to be less than 100 words.",
+      "model_config_name": "qwen_config",
+      "rag_config": {
```

Comment:
Can we do this pre-processing outside the agent? For example (taking `get` as example):
```python
knowledge_bank = KnowledgeBank(...)

knowledges = knowledge_bank.get(knowledge_ids=["kb1", "kb2"], similarity_top_k=5, log_retrieval=5, recent_n_mem=1)

AgentClass(name="assistant", knowledges=knowledges, ...)
```
or user can setup their own knowledge within the agent object's constructor by themselves. 

There are two advantages:
1. No need to know what parameters should be written in a rag config. All parameters are in the declaration of this `get()` function, which can be accessed easily.
2. The agent is not required to have a rag config attribute.

Reply:
Support after update. Now, the knowledge can be obtained by `get_knowledge` function, and a list of `knowledge` can be assigned to a agent in initialization.

In this update, agents are changed to use `knowledge.retrieve` function directly (the retriever is removed). The retriever is build in the `knowledge.retrieve` function every time called, with the parameter provided.